src/main/java/TurnUseCases/BuildUseCase/BuildBuildings.java

```diff
@@ -0,0 +1,149 @@
+package TurnUseCases.BuildUseCase;
+
+import GameEntities.Player;
+import GameEntities.Tiles.ColorPropertyTile;
+import GameEntities.Tiles.Property;
+
+import java.lang.Math;
+import java.util.ArrayList;
+
+
+/**
+ * All methods related to build buildings.
+ */
+
+public class BuildBuildings implements BuildBuildingInputBoundary{
+
+    /**
+     * Check if player can build buildings
+     *
+     * @param player the player who wants build buildings.
+     * @param property the property on which player wants to build buildings.
+     */
+    @Override
+    public boolean isBuildable (Player player, Property property){
+        if(!player.ownsProperty(property)){
+            return false;
+        }
+        if(property.isMortgaged()){
+            return false;
+        }
+        if(!property.getClass().getName().equals("GameEntities.Tiles.ColorPropertyTile")){
```

Comment:
If you're checking the class of the object, you can use `property instanceof ColorPropertyTile`